internal/command/jsonformat/differ/value.go

```diff
@@ -0,0 +1,91 @@
+package differ
+
+import (
+	"encoding/json"
+
+	"github.com/hashicorp/terraform/internal/command/jsonplan"
+)
+
+// Value contains the unmarshalled generic interface{} types that are output by
+// the JSON structured run output functions in the various json packages (such
+// as jsonplan and jsonprovider).
+//
+// A Value can be converted into a change.Change, ready for rendering, with the
+// ComputeChangeForAttribute, ComputeChangeForOutput, and ComputeChangeForBlock
+// functions.
+type Value struct {
```

Comment:
I'd like to bikeshed about the naming here a bit, because `Value` is very generic and seems a bit odd when it actually mostly represents a before-after pair.

In another plan rendering implementation, I used the (similarly vague) term `DiffItem`. That isn't any better, so let's not do that.

I'd suggest `Change` here, but you're already using that to represent what I think is a processed, renderable diff. Maybe that type and package should be considered for a rename too…?

Wooly, tentative suggestions:

- Consider renaming `Change` to something which indicates that it is a logical difference between current and planned state. For example, `Diff` or `Plan` or some awful compound involving nothingwords like `Item`, `Container`, or `Value`.
- Alternatively, since the `change` package is where renderers live, just take the shortcut and rename it to something about rendering or views.
- Consider renaming `Value` to something which indicates that it is an unprocessed pair of before and after values from current and planned state. For example, `Change`, `Delta`, etc.
- Before doing either of the above, if you haven't already, look for existing examples of similar concepts within the codebase which closely map to these concepts, and consider reusing those names.

(These suggestions of course do not block this PR merge.)

Reply:
Thanks for the suggestions, I've made a not of this and I will come back later and do a refactoring of the struct and package names 👍